modules/collect/src/main/java/com/opengamma/strata/collect/Guavate.java

```diff
@@ -69,6 +69,30 @@ private Guavate() {
 
   //-------------------------------------------------------------------------
   /**
+   * Uses a number of suppliers to create a single optional result.
+   * <p>
+   * The Java 8 {@link Optional} class does not have an {@code or} method,
+   * so this provides an alternative.
+   * 
+   * @param <T>  the type of element in the iterable
+   * @param suppliers  the suppliers to combine
+   * @return the list that combines the inputs
+   */
+  @SuppressWarnings("unchecked")
+  @SafeVarargs
+  public static <T> Optional<T> firstNotEmpty(Supplier<Optional<? extends T>>... suppliers) {
```

Comment:
The equivalent Guava method for null is called `firstNonNull` (note "non" instead of "not"). Would it be better to be consistent with that?